### ISSUE TITLE
wrong date is selected.

### ISSUE THREAD

**Issue by nakulpuranik** [Mulham-Raee/Horizontal-Calendar#37]
If the month format is passed as "MMMM" I'm not able to select the future date.
You can see that in GIF, I'm not able to select 8th Sep.
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/4466884/30104699-04c173c2-9314-11e7-8c17-f370815d67df.gif)

**Comment by BDeveloper** [comment]
Yes am also able to reproduce the same issue, and its happening when trying to use HorizontalCalenderView with custom layout. Just downloaded the sample and modified the .xml file, its reproducible. Check below mentioned sample i.e.

activity_main.xml

<?xml version="1.0" encoding="utf-8"?>
<RelativeLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:fitsSystemWindows="true"
    tools:context="devs.mulham.raee.sample.MainActivity">

    <android.support.design.widget.AppBarLayout
        android:id="@+id/appBar"
        android:layout_width="match_parent"
        android:layout_height="wrap_content">

        <android.support.v7.widget.Toolbar
            android:id="@+id/toolbar"
            android:layout_width="match_parent"
            android:layout_height="?attr/actionBarSize"
            android:background="?attr/colorPrimary"/>
    </android.support.design.widget.AppBarLayout>

    <LinearLayout
        android:id="@+id/leaderboard_layout"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:layout_weight="3"
        android:layout_margin="5dp"
        android:orientation="vertical"
        android:visibility="visible">

        <TextView
            android:id="@+id/UserName"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_gravity="center"
            android:layout_marginTop="10dp"
            android:layout_marginBottom="4dp"
            android:ellipsize="marquee"
            android:fontFamily="sans-serif"
            android:gravity="center"
            android:marqueeRepeatLimit="marquee_forever"
            android:textColor="#000000"
            android:textSize="18sp"
            android:textStyle="bold" />

        <TextView
            android:id="@+id/NumJoined"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_gravity="center"
            android:layout_marginTop="15dp"
            android:text="Software"
            android:textColor="@color/colorAccent"
            android:textSize="13sp" />

        <TextView
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_gravity="center"
            android:layout_marginLeft="2dp"
            android:layout_marginTop="10dp"
            android:text="Developer"
            android:textColor="@color/colorAccent"
            android:textSize="13sp" />

        <devs.mulham.horizontalcalendar.HorizontalCalendarView
            android:id="@+id/calendarView"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            app:textColorNormal="#bababa"
            app:textColorSelected="#FFFF"
            app:selectorColor="#c62828"
            app:sizeBottomText="14sp"
            app:sizeMiddleText="20sp"/>
    </LinearLayout>

        <android.support.design.widget.FloatingActionButton
            android:id="@+id/fab"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_alignParentBottom="true"
            android:layout_alignParentRight="true"
            android:layout_margin="@dimen/fab_margin"
            android:src="@drawable/ic_calendar_24dp"/>
</RelativeLayout>

MainActivity.java

Calendar startDate = Calendar.getInstance();
startDate.add(Calendar.DATE, -2);

Calendar endDate = Calendar.getInstance();
endDate.add(Calendar.DATE, 2);

final Calendar defaultSelectedDate = Calendar.getInstance();

horizontalCalendar = new HorizontalCalendar.Builder(this, R.id.calendarView)
		.range(startDate, endDate)
		.datesNumberOnScreen(5)
		.configure()
			.formatTopText("MMM")
			.formatMiddleText("dd")
			.formatBottomText("EEE")
			.showTopText(true)
			.showBottomText(true)
			.textColor(Color.LTGRAY, Color.WHITE)
			.colorTextMiddle(Color.LTGRAY, Color.parseColor("#ffd54f"))
		.end()
		.defaultSelectedDate(endDate)
		.build();

Kindly let me know what need to be done for this issue, as am stuck here.

**Comment by sam-androiddev** [comment]
Not able to select yesterday's date. Although no relation to MMMM format in my case. default sample is not working. This is how I am using it:

<devs.mulham.horizontalcalendar.HorizontalCalendarView
                    android:id="@+id/calendarView"
                    android:layout_width="match_parent"
                    android:layout_height="wrap_content"
                    android:padding="@dimen/spacing_micro"
                    android:background="@color/colorPrimary"
                    app:textColorSelected="#FFFF"/>



/* starts before 1 month from now */
        Calendar startDate = Calendar.getInstance(TimeZone.getDefault());
        startDate.add(Calendar.MONTH, -1);
        /* ends after 1 month from now */
        Calendar endDate = Calendar.getInstance(TimeZone.getDefault());
        endDate.add(Calendar.MONTH, 1);

        horizontalCalendar = new HorizontalCalendar.Builder(this, R.id.calendarView)
                .range(startDate, endDate)
                .datesNumberOnScreen(5)
                .build();

**Comment by gswebtech7** [comment]
Is the issue resolved ? I am also trying to select the current date but the same issue occurs. Its is showing unselectable

**Comment by gswebtech7** [comment]
Is the issue resolved ? I am also trying to select the current date but the same issue occurs. Its is showing unselectable

**Comment by gswebtech7** [comment]
Is the issue resolved ? I am also trying to select the current date but the same issue occurs. Its is showing unselectable
